### PR TITLE
Twitter oAuth complete

### PR DESCRIPTION
**Pull request by coutvv**

### CODE REVIEW

File: src/main/java/ru/scv3/controller/auth/TwitterAuthController.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+        String oauthToken = request.queryMap("oauth_token").value();
+        if(oauthToken == null) {
+            String requestToken = TwitterAuthUtil.getRequestToken();
+            String link = TwitterAuthUtil.getLinkForAuth(requestToken);
+            response.redirect(link);
+        } else {
+            String verifier = request.queryMap("oauth_verifier").value();
+            String token = TwitterAuthUtil.getAccessToken(oauthToken, verifier);
+            getUserData(token);
+            User user = getOrCreateUser(getUserData(token));
```
Comment (Som3) [inline src/main/java/ru/scv3/controller/auth/TwitterAuthController.java]: dublication

File: src/main/java/ru/scv3/controller/auth/TwitterAuthController.java [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+            AuthUtil.signIn(request, response, user);
+        }
+        response.redirect("#");
+        return null;
+    }
+
+    private UserData getUserData(String token){
+        String id = (String) TwitterAuthUtil.parseToken(token).get("user_id");
+        String screenName = (String) TwitterAuthUtil.parseToken(token).get("screen_name");
```
Comment (Som3) [inline src/main/java/ru/scv3/controller/auth/TwitterAuthController.java]: may be save result parseToken?

File: src/main/java/ru/scv3/util/TwitterAuthUtil.java [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+                .setHost("api.twitter.com")
+                .setPort(-1)
+                .setPath("/oauth/access_token")
+                .setParameters(qparams).build();
+
+
+        return doPostRequest(uri);
+    }
+
+    public static Properties parseToken(String token) {
```
Comment (Som3) [inline src/main/java/ru/scv3/util/TwitterAuthUtil.java]: why prop? he sync, i think map is better

File: src/main/java/ru/scv3/util/TwitterAuthUtil.java [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+        String[] props = token.split("&");
+        for(String prop : props) {
+            String[] pair = prop.split("=");
+            String name = pair[0]; String value = pair[1];
+            result.put(name , value);
+        }
+        return result;
+    }
+
+    private static String doPostRequest(URI uri) throws IOException {
```
Comment (Som3) [inline src/main/java/ru/scv3/util/TwitterAuthUtil.java]: create new instance httpclient by each request? u sure?

Comment (coutvv) [inline src/main/java/ru/scv3/util/TwitterAuthUtil.java]: twitter api demand it

**Pull request merged**